mvp+jest_winston

mixmaster1989 opened the pull request.

mixmaster1989 committed:
внедряюснова pdf

mixmaster1989 committed:
продолжаем внедрять пдф

mixmaster1989 committed:
убраны все упоминания webapp, сунут pdf везде

mixmaster1989 committed:
начинает бесить это внедрение...

mixmaster1989 committed:
предпросмотр в выводе

mixmaster1989 committed:
снова предпросмотр

mixmaster1989 committed:
сука

mixmaster1989 committed:
ссссукааааа

mixmaster1989 committed:
тайминг генера теста

mixmaster1989 committed:
тайминг внедрен. исправление вывод вариантов

mixmaster1989 committed:
прогрессбар

ThePositree committed:
WIP

mixmaster1989 committed:
аш комментарий

mixmaster1989 committed:
исправление ошибок

mixmaster1989 committed:
чек изменений

mixmaster1989 committed:
откат

mixmaster1989 committed:
исправление синтаксиса в модуле доктупдф

mixmaster1989 committed:
мердж

mixmaster1989 committed:
оыау0аыа

mixmaster1989 committed:
щалзку

mixmaster1989 committed:
ппавпвап

mixmaster1989 committed:
онтокен

mixmaster1989 committed:
cnbvbyu

mixmaster1989 committed:
streaming

mixmaster1989 committed:
tot ghj,s

mixmaster1989 committed:
откат

mixmaster1989 committed:
спросить ии

mixmaster1989 committed:
логи

mixmaster1989 committed:
еще

mixmaster1989 committed:
общение с ии

mixmaster1989 committed:
диалог с ии

mixmaster1989 committed:
генер кеша 2.0

mixmaster1989 committed:
тезисы

mixmaster1989 committed:
промт

mixmaster1989 committed:
физзи серч

mixmaster1989 committed:
логи кеша

mixmaster1989 committed:
кеш в ЯД

mixmaster1989 committed:
переход на модульность

mixmaster1989 committed:
орректировка

mixmaster1989 committed:
ВНЕДРЕНИЕ JEST

mixmaster1989 committed:
коммит

mixmaster1989 committed:
удаление цункции кеша из основного модуля

mixmaster1989 committed:
корректировка яд+json

mixmaster1989 committed:
внедрение записи кеша на ЯД+тест сразу

mixmaster1989 committed:
ребилд теста кеш+яд

mixmaster1989 committed:
ребилд теста

mixmaster1989 committed:
внедрение модуля сообщений пользователю+ тест

mixmaster1989 committed:
чистка проекта от левых файлов

mixmaster1989 committed:
jest на файл bot.js через МОК+ вывод тезисов после каждого блока генерации

mixmaster1989 committed:
правка модуля пдф для прохождения теста

mixmaster1989 committed:
переделка теста

mixmaster1989 committed:
Внедрение winston часть первая

mixmaster1989 committed:
внедрение winston часть 2 + ПОЛНАЯ обвязка JEST

mixmaster1989 committed:
исправление ошибок внедрения jest

mixmaster1989 committed:
test: корректная асинхронная обработка и моки для unit-тестов

mixmaster1989 committed:
удаление следов

mixmaster1989 committed:
правим экспорт

mixmaster1989 committed:
правим тесты снова

mixmaster1989 committed:
еще правим тесты2

mixmaster1989 committed:
последний тест исправляем

ThePositree commented:
Не хватает описания в README.md, нужно описать, что за сервис, зачем нужен, на каких технологиях написан, какие npm-скрипты поддерживает (их можно найти в package.json под ключом «scripts») и как запустить локально.

ThePositree commented:
fuzzysort библиотека не индексируется как зависимость в package.json, надо указать что это зависимость сервиса (npm install fuzzysort)

ThePositree commented on services/yadisk_service.js:
```diff
@@ -45,6 +46,18 @@ class YaDiskService {
+            logger.info(`[YaDisk/${operation}] ${message}`);
+        }
+
+        // Оставим и консоль для наглядности (опционально)
         console.log(`${emoji[level] || '🔄'} [${timestamp}] [YaDisk/${operation}] ${message}${error ? `\n  Error: ${JSON.stringify(logEntry.error, null, 2)}` : ''}`);
```
Лучше убрать console.log, сообщения в консоле дублируются и вводят в заблуждение.

ThePositree commented on services/yadisk_service.js:
```diff
@@ -45,6 +46,18 @@ class YaDiskService {
+            logger.info(`[YaDisk/${operation}] ${message}`);
+        }
+
+        // Оставим и консоль для наглядности (опционально)
```
Советую вообще убрать log метод у этого класса и логировать всё на самых верхних уровнях кода (то есть в нашем случае в файле bot.js). Внедрять логирование в модули приложения — не очень хорошая идея, нужно стараться, чтобы функции работали без сайд-эффектов (ведение логов в функции — это один из сайд-эффектов).

ThePositree commented on bot.js:
```diff
@@ -1,240 +1,481 @@
+const { convertDocxToPdf } = require("./modules/docx2pdf");
+const { saveToCacheHistory, getAllCacheQuestions, fuzzyFindInCache } = require("./modules/cache");
+
+const yadisk = new YaDiskService(process.env.YANDEX_DISK_TOKEN);
```
Перед тем как прокинуть токен в этот модуль, проверь токен на валидность (что он существует и что это строка), не нужно перекидывать функционал валидации модулям, которые отвечают за совсем другие действия, иначе эти модули превратятся в класс бога (это такой антипаттерн в программировании), класс бога может быть только главный файл, который отвечает за старт процесса, и то если проект маленький, в больших проектах можно было бы создать отдельный модуль, который отвечает за валидацию сущностей в проекте.

ThePositree commented on bot.js:
```diff
@@ -1,240 +1,481 @@
+
+const yadisk = new YaDiskService(process.env.YANDEX_DISK_TOKEN);
+
+const PORT = process.env.PORT || 3000;
+const BOT_TOKEN = process.env.BOT_TOKEN;
+const modelName = "Nous-Hermes-2-Mistral-7B-DPO.Q4_0.gguf";
+const modelDir = path.join(os.homedir(), ".cache", "gpt4all");
+const materialsPath = path.join(__dirname, "materials");
+const cachePath = path.join(__dirname, "cache");
+const tempPath = path.join(__dirname, "temp");
+const gpt4allPath = path.join(modelDir, modelName);
+const gpt4allCachePath = path.join(gpt4allPath, "cache");
```
Создай модуль config, в котором будут все твои константы, а также env, то есть в этом модуле ты сделаешь dotenv.config и заберешь в константы env переменные. Это будет единое место, в котором собрана вся конфигурация сервиса.

ThePositree commented on modules/cache.js:
```diff
@@ -0,0 +1,52 @@
+const fuzzysort = require("fuzzysort");
+const logger = require("./logger");
+
+const db = new sqlite3.Database(path.join(__dirname, "..", "database.sqlite"));
+
```
Сюда же перенеси логику создания таблиц из bot.js.

ThePositree commented:
Так же установи как зависимость jest

ThePositree commented:
добавить папку logs в gitignore, логи не должны быть в репозитории.

ThePositree commented on bot.js:
```diff
@@ -1,240 +1,481 @@
+const express = require("express");
+const fs = require("fs-extra");
+const path = require("path");
+const os = require("os");
```
Удалить неиспользуемые импорты и константы.

ThePositree commented on package.json:
```diff
@@ -4,29 +4,31 @@
     "pm2:stop": "pm2 stop telegram-bot",
     "audit": "npm audit fix --force",
-    "clean": "rimraf node_modules package-lock.json && npm cache clean --force"
+    "clean": "rimraf node_modules package-lock.json && npm cache clean --force",
```
удалять package-lock.json нельзя, этот файл нужен для того что бы следить за версиями установленных библиотек и что бы у всех разработчиков были установлены одинаковые версии зависимостей.

ThePositree commented on bot.js:
```diff
@@ -1,240 +1,481 @@
+const gpt4allCachePath = path.join(gpt4allPath, "cache");
+const bot = new Telegraf(BOT_TOKEN);
+const app = express();
+app.use("/static", express.static(path.join(__dirname, "static")));
```
Тут ты разворачиваешь express http сервер, но я не вижу где он используется потом. Зачем в этом проекте express?

ThePositree commented on bot.js:
```diff
@@ -1,240 +1,481 @@
+}
+
+// Обработка ошибок при генерации вопросов
+async function generateAIQuestions(text) {
+  try {
+    const maxInputLength = 700;
+    const truncatedText = text.length > maxInputLength ? text.substring(0, maxInputLength) + "..." : text;
+    const prompt = `Сформулируй один короткий вопрос с 4 вариантами ответов по тексту ниже. Отметь правильный вариант.\nВОПРОС:\nА)\nБ)\nВ)\nГ)\nПРАВИЛЬНЫЙ:`;
+    if (!gpt4allModel) gpt4allModel = await initGPT4AllModel();
+
+    return await gpt4allModel.generate(`${prompt}\n\n${truncatedText}`);
+  } catch (error) {
+    logger.error("Ошибка при генерации вопросов: " + error.message);
+    throw new Error("Не удалось сгенерировать вопросы.");
+  }
+}
+
+function parseTestResponse(response) {
```
Удалить неиспользуемые функции.

Commit:
переход на ламу

Commit:
возврат на мистраль

Commit:
правка джест теста

Commit:
внедрение ocr этап 1

Commit:
внедрение ocr этап 2

Commit:
test-ocr.png

Commit:
исправление модуля оср

Commit:
исправление модуля оср2

Commit:
исправление модуля оср3

Commit:
исправление модуля оср4

Commit:
исправление модуля оср5

Commit:
исправление модуля оср6

Commit:
улучшение предобработки

Commit:
улучшение предобработки(opencv)

Commit:
установка питона

Commit:
настройка предобработки

Commit:
правка теста

Commit:
внедрение кирилицы

Commit:
удаление тессеракта, какашка вонючая на деле

Commit:
внедрение paddleocr

Commit:
внедрение paddleocr1

Commit:
внедрение paddleocr2

Commit:
внедрение paddleocr3

Commit:
предобработка

Commit:
постобработка

Commit:
постобработка тесты

Commit:
постобработка тесты2

Commit:
постобработка тесты3

Commit:
постобработка тесты4

Commit:
тюнинг пред и постобработки по результатам тестов

Commit:
внедрение ии на предпост обработку

Commit:
внедрение ии на предпост обработку2

Commit:
внедрение ии на предпост обработку3

Commit:
внедрение ии на предпост обработку4

Commit:
внедрение ии на предпост обработку5

Commit:
внедрение ии на предпост обработку6

Commit:
внедрение ии на предпост обработку7

Commit:
внедрение ии на предпост обработку8

Commit:
внедрение ии на предпост обработку9

Commit:
внедрение ии на предпост обработку10

newtrygit committed:
откат. без ии на постпред обработке. на данный момент к понцепте бота не реализуемо. оставляем только paddle

newtrygit committed:
падле+тест

newtrygit committed:
внедрение 9 тестовых шаблонов для фидбека

newtrygit committed:
внедрение новых шаблонов на основе фидбека от 9 прошлых

newtrygit committed:
внедрение новых шаблонов на основе фидбека от 9 прошлых 2

newtrygit committed:
внедрение новых шаблонов на основе фидбека от 9 прошлых 3

newtrygit committed:
внедрение кропа

newtrygit committed:
внедрение кропа2

newtrygit committed:
посткоррекция

newtrygit committed:
посткоррекция8

newtrygit committed:
посткоррекция9

newtrygit committed:
посткоррекция10

newtrygit committed:
посткоррекция11

newtrygit committed:
посткоррекция12

newtrygit committed:
посткоррекция13

newtrygit committed:
посткоррекция14

newtrygit committed:
посткоррекция15

newtrygit committed:
посткоррекция16

newtrygit committed:
посткоррекция17

newtrygit committed:
посткоррекция18

newtrygit committed:
посткоррекция19

newtrygit committed:
посткоррекция20

newtrygit committed:
посткоррекция21

newtrygit committed:
посткоррекция22

newtrygit committed:
посткоррекция23

newtrygit committed:
посткоррекция24

newtrygit committed:
посткоррекция25

newtrygit committed:
новый уровень! 10 шаблонов+анализ=вывод заготовки под семантику! стадия 1

newtrygit committed:
новый уровень! 10 шаблонов+анализ=вывод заготовки под семантику! стадия 2

newtrygit committed:
новый уровень! 10 шаблонов+анализ=вывод заготовки под семантику! стадия 3

newtrygit committed:
новый уровень! 10 шаблонов+анализ=вывод заготовки под семантику! стадия 4

newtrygit committed:
новый уровень! 10 шаблонов+анализ=вывод заготовки под семантику! стадия 5

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 1

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 2

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 3

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 4

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 5

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 6

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 7

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 8

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 8.1

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 8.2

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 9

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 9.1 логи

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 10

newtrygit committed:
СЕМАНТИКА! предобработка, стадия 10недрена улучшенная формула оценки человекочитаемости и полезности OCR-результата (evalHumanReadableScoreV2). Теперь учитываются телефоны, email, ИНН, адреса, бизнес-термины, разнообразие строк, а также фильтрация мусорных строк. В логах выводится подробная таблица сравнения всех вариантов, и пользователю отправляется самый полезный и человекочитаемый текст.

newtrygit committed:
семантика v 2.0

newtrygit committed:
семантика v 3.0

newtrygit committed:
семантика v 3.1

newtrygit committed:
клавиатуру чиним, с утра

newtrygit committed:
стейбл версия. делаем логирование функций постобработки.

newtrygit committed:
OCR-логика теперь полностью вынесена в отдельные модули и интегрирована в bot.js:

Импортированы getTemplates и processOcrPipeline из modules/ocr.
В обработчике ocr_all_templates теперь вызывается только processOcrPipeline, а не локальные функции и ручная обработка.
Старые определения ocrTemplates, evalHumanReadableScore, selectBestOcrResult, sendBestOcrResult, filterGarbage, addGarbage и ручная postprocess-цепочка удалены из bot.js.
Все обращения к шаблонам теперь централизованы через getTemplates().

newtrygit committed:
модульный ocr, патч 0.1

newtrygit committed:
модульный ocr, патч 0.2  вынес функции semanticOcrAssemble и humanReadableAssemble в отдельный модуль modules/ocr/postprocess.js, экспортировал их и заменил импорты в bot.js на корректные.

                        Что исправлено:

                        Ошибка Cannot find module './semanticOcrAssemble' устранена: теперь используется require('./modules/ocr/postprocess').
                        Код стал чище и модульнее: логика постобработки и сборки результатов OCR теперь централизована.

newtrygit committed:
перь при получении нового фото бот сбрасывает состояние сравнения оригинала (userStates[userId] и userLastOcr[userId]), если пользователь был в режиме "awaiting_original". Это устраняет баг: после ввода оригинала и отправки новой картинки бот снова реагирует корректно.

Что исправлено:

После сравнения оригинала можно сразу отправлять новое фото — бот не будет "молчать", а предложит снова выбрать способ обработки OCR.
Проверьте, что теперь бот реагирует на повторные картинки без зависаний!

newtrygit committed:
зафиксирована stable версия продукта. актуализация старой обвязки jest и новые тесты.

newtrygit committed:
нуукненеунекнкенен

newtrygit committed:
аккуратно

newtrygit committed:
ыфвавыфа

newtrygit committed:
убираем семантику

newtrygit committed:
Внедряем новую пост обработку

newtrygit committed:
логи конечки

newtrygit committed:
избавляемся от дублей

newtrygit committed:
еще убираем дубли

newtrygit committed:
внедряем string-similarity

newtrygit committed:
агрессивное сравнение

newtrygit committed:
еще попытка

newtrygit committed:
исправляем экранирование

newtrygit committed:
чиним ЯД

newtrygit committed:
чиним ЯД 2

newtrygit committed:
причесали комменты

newtrygit committed:
МЕГА АККУРАТНО!!! ЕСЛИ ОТКАТ ТО НА ПРЕДЫДУЩИЙ КОММИТ!!!!!

newtrygit committed:
внедряем тесты

newtrygit committed:
внедряем тесты 2

newtrygit committed:
тесты 3

newtrygit committed:
тесты 4

newtrygit committed:
тесты 5

newtrygit committed:
тесты 6

newtrygit committed:
тесты 7

newtrygit committed:
тесты 8

newtrygit committed:
тест ЯДа послан нахуй. временно.

newtrygit committed:
переход на мамзон q

newtrygit committed:
тесты почти готовы

newtrygit committed:
обвязка уже зеленая но тесты подстроены под код, делаем тесты реально проверяющими работу. этап 1

newtrygit committed:
этап 2

newtrygit committed:
этап 3

newtrygit committed:
этап 4

newtrygit committed:
этап 5

newtrygit committed:
этап 6

newtrygit committed:
этап 7

newtrygit committed:
этап 8

newtrygit committed:
этап 9

newtrygit committed:
этап 10

newtrygit committed:
удаляем гавнотест и гавнофайл

newtrygit committed:
снова правки тестов
